run.go

```diff
@@ -86,16 +120,142 @@ func createGrpcServer() *grpc.Server {
 	return service.GRPCServer
 }
 
-func startPrometheusServer() {
+func startPrometheusServer(started func()) {
 	service.PrometheusServer = &http.Server{Addr: service.PrometheusConfig.Address()}
 
 	http.Handle("/metrics", promhttp.Handler())
 	logrus.Infof("Prometheus metrics at http://%s/metrics", service.PrometheusConfig.Address())
 
+	service.prometheusRunning = true
+	started()
+
 	go func() {
 		if err := service.PrometheusServer.ListenAndServe(); err != nil {
```

Comment:
Should the service marked as running only after the ListenAndServe() is called successfully?

Reply:
ListenAndServe is blocking from what I can tell. Could call Listen which gets us our port then call started() then Serve()? Still wouldn't be 100% true but close enough?

Reply:
split into listen and serve